### PR TITLE
=htp #20884 fix header name rendering issue when $minus present

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/20884

### CODE REVIEW

**Comment by ktoso** [comment]
PLS BUILD

**Comment by johanandren** [comment]
LGTM!

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala [hunk 1]
```diff
@@ -36,6 +36,14 @@ sealed abstract class ModeledCompanion[T: ClassTag] extends Renderable {
       case res                                         ⇒ Left(res.errors)
     }
 }
+/** INTERNAL API */
+private[akka] object ModeledCompanion {
+  def nameFromClass[T](clazz: Class[T]): String = {
+    val name = clazz.getSimpleName.replace("$minus", "-")
```
Comment (viktorklang) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala]: getSimpleName has problems in Scala.

Use: https://github.com/akka/akka/blob/master/akka-actor/src/main/scala/akka/event/Logging.scala#L386

Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala]: Hah, good catch - will fix.

**Pull request merged**